### PR TITLE
Output: use logging, remove verbose output by default

### LINKED ISSUE #489: Lower verbosity by default or add option to allow removing prefixed and suffixed debug output
**Is your feature request related to a problem? Please describe.**
podman-compose prints additional debug information when commands are executed, and there appears to be no way to silence it. Some tools process CLI output as-is and expect it to be in a certain format (such as JSON), but the additional output printed by podman-compose before the actual command output can break these tools.

**Describe the solution you'd like**
Perhaps lower the default verbosity to not print debug output unless `-v` or `--verbose` is passed.

**Describe alternatives you've considered**
Leaving the current behaviour the same, and adding a flag `--silent`, `--verbosity=none` or similar, to achieve the same effect.

**Additional context**
The version I'm using is 1.0.3.

Below is an example of the problem. Suppose I have some tool that wants to execute `echo hi`, and I want to wrap this in a container using Podman, I override the command to be:

```
podman exec --interactive --tty some_container_1 echo hi
```

The output is:

```
hi
```

Great, that works. Now I want to use podman-compose to do the same, and execute the command from an existing container in my pod, so I turn this into:

```
podman-compose exec some_container echo "hi"
```

The output is now the expected output, prefixed and suffixed with debug information:

```
['podman', '--version', '']
using podman version: 4.0.3
podman exec --interactive --tty some_container_1 echo hi
hi
exit code: 0
```

This debug information is useful, but is usually redundant in scenarios such as `podman-compose up` (and **also seems to expose secrets** if you pass them through the environment, as it also logs the environment variables it passes), so the best approach seems to lower the verbosity by default.

### PR DESCRIPTION
**Pull request by maurerle**
Fixes #489.

Introduces a --verbose flag if you want to see all the noise that was previously printed by default.

### CODE REVIEW

**Comment by maurerle** [comment]
I made the linter happy, so this can be merged

**Comment by maurerle** [comment]
@muayyad-alsadi can I do anything to help getting this merged?

**Comment by crdoconnor** [comment]
This would be extremely useful.

**Comment by errodrigues** [comment]
+1
Any plans to include this in the next tag/release? Would be a game changer for QoL!

**Comment by nassir90** [comment]
I too hope this can be merged soon. It's a pretty minor code change for a massive QoL improvement.

**Comment by maurerle** [comment]
Hi @ParetoOptimalDev,
i think there is also the possibility to use docker-compose with podman - instead of the unmaintained podman-compose version.
See here:
https://docs.podman.io/en/latest/markdown/podman-compose.1.html

Other than that you can try my fork of podman-compose which has merged a lot of the missing features which are available in open pull requests here:
https://github.com/maurerle/podman-compose

**Comment by thiagolsfortunato** [comment]
any news? 👀

**Comment by crdoconnor** [comment]
I think this project is abandoned.

File: podman_compose.py [hunk 1]
```diff
@@ -394,7 +392,7 @@ def assert_volume(compose, mount_dict):
     proj_name = compose.project_name
     vol_name = vol["name"]
     is_ext = vol.get("external", None)
-    log(f"podman volume inspect {vol_name} || podman volume create {vol_name}")
+    log.debug("podman volume inspect %s || podman volume create %s", vol_name, vol_name)
```
Comment (p12tic) [inline podman_compose.py:395]: Is there a way to keep f-string formatting here?

**Review by p12tic — APPROVED** [review]
Looks great, thank you!

The PR will need a rebase to latest main branch.

**Comment by p12tic** [comment]
cc @maurerle or anyone else who's interested. This PR could easily land if someone rebased it.

File: podman_compose.py (hunk 1 above)
Comment (maurerle) [reply podman_compose.py:395]: Thank you for your review!
There is no better approach to this as I know.
It is best practice to lazy eval in logging (which is currently only supported with % placeholders) but it is efficient to use f-strings everywhere else.

**Comment by maurerle** [comment]
Thanks @p12tic ! hopefully this will get merged finally :)

**Comment by p12tic** [comment]
We may need to re-add part of logging that prepended task name to log output. However this is useful only to podman-compose developers. There are many more users who will find their output much cleaner now. So it makes sense to postpone further improvements.

**Pull request merged**